src/python/pants/engine/fs.py

```diff
@@ -96,15 +104,15 @@ class DigestContents(Collection[FileContent]):
     """
 
 
-class DigestEntries(Collection[Union[FileEntry, Directory]]):
+class DigestEntries(Collection[Union[FileEntry, SymlinkEntry, Directory]]):
     """The indirect file contents of a Digest.
 
     DigestEntries is a collection of FileContent and Directory instances representing, respecively,
```

Comment:
Both of these types need doc updates. It might also be worth updating the docstring for `Snapshot` to point out the contrast with these types (symlink oblivious vs aware).